# Patch to fix compilation issues with FFADO on Slackare 13.37
# Copyright (C) 2012 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
# 
# $Id: libffado-2.1.0.diff,v 1.1 2012/09/17 14:24:56 gsotirov Exp $
# 

diff -urNad libffado-2.1.0-orig/src/bebob/bebob_dl_mgr.cpp libffado-2.1.0/src/bebob/bebob_dl_mgr.cpp
--- libffado-2.1.0-orig/src/bebob/bebob_dl_mgr.cpp	2012-07-06 13:31:48.000000000 +0300
+++ libffado-2.1.0/src/bebob/bebob_dl_mgr.cpp	2012-09-17 14:55:57.000000000 +0300
@@ -35,6 +35,7 @@
 
 #include <cstdio>
 #include <cstring>
+#include <memory>
 #include <unistd.h>
 
 namespace BeBoB {
diff -urNad libffado-2.1.0-orig/src/ffadodevice.h libffado-2.1.0/src/ffadodevice.h
--- libffado-2.1.0-orig/src/ffadodevice.h	2012-03-08 13:27:34.000000000 +0200
+++ libffado-2.1.0/src/ffadodevice.h	2012-09-17 14:55:57.000000000 +0300
@@ -34,6 +34,7 @@
 
 #include <vector>
 #include <string>
+#include <memory>
 
 class DeviceManager;
 class ConfigRom;
diff -urNad libffado-2.1.0-orig/src/libutil/serialize_expat.cpp libffado-2.1.0/src/libutil/serialize_expat.cpp
--- libffado-2.1.0-orig/src/libutil/serialize_expat.cpp	2009-12-19 16:12:52.000000000 +0200
+++ libffado-2.1.0/src/libutil/serialize_expat.cpp	2012-09-17 14:56:36.000000000 +0300
@@ -21,8 +21,10 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  *
  */
- // FOR CACHE_VERSION
 
+#include <cstdlib>
+
+#include "version.h" // FOR CACHE_VERSION
 #include "serialize.h"
 
 using namespace std;
